docs/plugins/composability.md

```diff
@@ -0,0 +1,585 @@
+top-level `routes.ts`. This is to avoid circular imports when you use the route
+references from other parts of the app.
+
+### Parameterized Routes
+
+A new addition to `RouteRef`s is the possibility of adding named and typed
+parameters. Parameters are declared at creation, and will enforce presence of
+the parameters in the path in the app, and require them as a parameter when
+using `useRouteRef`.
+
+The following is an example of creation and usage of a parameterized route:
+
+```tsx
+// Creation of a parameterized route
+const myRouteRef = createRouteRef({
+  title: 'My Named Route',
+  params: ['name']
+})
+
+// In the app, where MyPage is a routable extension with myRouteRef set as mountPoint
+<Route path='/my-page/:name' element={<MyPage />}/>
+
+// Usage within a component
+const myRoute = useRouteRef(myRouteRef)
+return (
+  <div>
+    <a href={myRoute({name: 'a'})}>A</a>
+    <a href={myRoute({name: 'b'})}>B</a>
+  </div>
+)
+```
+
+It is currently not possible to have parameterized `ExternalRouteRef`s, or to
+bind an external route to a parameterized route, although this may be added in
+the future if needed.
+
+### New Catalog Components
+
+The established pattern for selecting what plugins should be available on each
+catalog page is to use custom components in the app, with logic embedded in the
+render function. Typically this takes form as a component that either receives
+the entity via props or uses the `useEntity` hook to retrieve the selected
+entity. A `switch` or `if` / `else if` chain is then used to select what
+children should be rendered based on information in the entity.
+
+This pattern will no longer work with the new composability system, and in
+general is very difficult to build any form declarative model around, as it
+depends on runtime execution. To help replace existing code, a new
+`EntitySwitch` component has been added to the `@backstage/catalog` plugin,
+which grabs the selected entity from context, and selects at most one element to
+render using a list of `EntitySwitch.Case`s children.
+
+For example, if you want all entities of kind `"Template"` to be rendered with a
+`MyTemplate` component, and all other entities to be rendered with a `MyOther`
+component, you would do the following:
+
+```tsx
+<EntitySwitch>
+  <EntitySwitch.Case if={isKind('template')}>
+    <MyTemplate />
+  </EntitySwitch.Case>
+
+  <EntitySwitch.Case>
+    <MyTemplate />
+  </EntitySwitch.Case>
+</EntitySwitch>
+
+// Shorter form if desired:
+<EntitySwitch>
+  <EntitySwitch.Case if={isKind('template')} children={<MyTemplate />}/>
+  <EntitySwitch.Case children={<MyTemplate />}/>
+</EntitySwitch>
+```
+
+The `EntitySwitch` component will render the children of the first
+`EntitySwitch.Case` that returns `true` when the selected entity is passed to
+the function of the `if` prop. If none of the cases match, no children will be
+rendered, and if a case doesn't specify an `if` filter function, it will always
```

Comment:
Does this effectively act as a default / fallback case?

Reply:
Yep! But it's not required, in case you don't want to render anything if there are no matches